### PR TITLE
cmake cleanup in CCPP; cmake in FV3dycore

### PR DESCRIPTION
**Pull request by aerorahul**
## Description

This PR:
- removes the CMake build of the dycore from fv3atm
- places the CMake build of the dycore in the dycore repo.  See [GFDL_atmos_cubed_sphere PR 141](https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/141)
- Uses installed packages of CCPP from  https://github.com/NCAR/ccpp-framework/pull/382 and https://github.com/NCAR/ccpp-physics/pull/705

## Testing

How were these changes tested?  RT's
What compilers / HPCs was it tested with?  RT's
Are the changes covered by regression tests? Yes.
Have the ufs-weather-model regression test been run? On what platform?  In progress
- Will the code updates change regression test baseline? No
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/141
- waiting on https://github.com/NCAR/ccpp-framework/pull/382
- waiting on https://github.com/NCAR/ccpp-physics/pull/705

### CODE REVIEW

File: ccpp/CMakeLists.txt [hunk 1]
```diff
@@ -96,10 +92,7 @@ endif()
 add_subdirectory(framework)
 
 add_subdirectory(physics)
-add_dependencies(ccppphys ccpp)
-target_link_libraries(ccppphys PUBLIC w3nco::w3nco_d NetCDF::NetCDF_Fortran)
-# This should not be necessary once framework and physics targets define BUILD_INTERFACE
-target_include_directories(ccppphys PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/framework/src)
+#add_dependencies(ccpp_physics ccpp_framework) ! Not sure if this is needed at all
```
Comment (climbfuji) [inline ccpp/CMakeLists.txt]: If this line is not needed, then please remove it.

Comment (aerorahul) [reply ccpp/CMakeLists.txt]: done in a89df7b

**Comment by junwang-noaa** [comment]
Do we have CCPP PRs for this fv3 PR? can we list them in the Dependencies section?

**Comment by aerorahul** [comment]
> Do we have CCPP PRs for this fv3 PR? can we list them in the Dependencies section?

As soon as @climbfuji opens the relevant PR's, I will update the description here.

**Comment by climbfuji** [comment]
> > Do we have CCPP PRs for this fv3 PR? can we list them in the Dependencies section?
> 
> As soon as @climbfuji opens the relevant PR's, I will update the description here.

These PRs were opened many weeks ago ...

Associated PRs (see also ufs-weather-model PR):

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/141 
https://github.com/NCAR/ccpp-framework/pull/382
https://github.com/NCAR/ccpp-physics/pull/705
https://github.com/NOAA-EMC/fv3atm/pull/385
https://github.com/ufs-community/ufs-weather-model/pull/791

**Comment by aerorahul** [comment]
> Do we have CCPP PRs for this fv3 PR? can we list them in the Dependencies section?

done.

**Review by junwang-noaa — APPROVED** [review]

**Review by climbfuji — CHANGES_REQUESTED** [review]
This PR, together with https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/141, changed the regression test results on AWS (CI) and Jet with Intel for yet unknown reasons.

**Comment by aerorahul** [comment]
The atmosphere dycore build is treated differently than the other components of the UFS.
Isolating and fixing this issue will likely result in non-reproducible baseline due to this special handling.

**Pull request closed without merging**